## Question 1
For each of the following expressions, what value will the expression give? Verify your answers by typing the expressions into Python or running them on Colab.
<pre>
1.   9 - 3 = 6
2.   8 * 2.5 = 20.0
3.   9 / 2 = 4.5
4.   9 / -2 = -4.5
5.   9 // -2 = -5
6.   9 % 2 = 1
7.   9.0 % 2 = 1.0
8.   9 % 2.0 = 1.0
9.   9 % -2 = -1
10.   -9 % 2 = 1
11.   9 / -2.0 = -4.5
12.   4 + 3 * 5 = 19
13.   (4 + 3) * 5 = 35

</pre>

## Question 2

1. Is the following line prints True: *No*


```
print(type(3) == type("3"))
```

2. Is the following line prints True: *Yes*



```
print(str(3) == "3")
```




## Question 3

Write two assignment statements that do the following:

1. Create a new variable, `temp`, and assign it the value 24.

2. Convert the value in `temp` from Celsius to Fahrenheit by multiplying by 1.8 and adding 32; make `temp` refer to the resulting value.

What is `temp`’s new value?

In [1]:
## Your code and answer
temp = 24
temp *= 1.8 ;temp += 32
temp

75.2

Temp's new value is 75.2

## Question 4

For each of the following expressions, in which order are the subexpressions evaluated?

<pre>
1.    6 * 3 + 7 * 4
2.    5 + 3 / 4
3.    5 - 2 * 3 ** 4
</pre>

1. `6 * 3 + 7 * 4`
- Step 1: Multiplications are performed first. So `6 * 3` is evaluated to 18 and `7 * 4` is evaluated to `28`.
- Step 2: Additions are done next. `18 + 28` results in `46`.
2. `5 + 3 / 4`
- Step 1: Division `3 / 4` is performed first, yielding `0.75`.
- Step 2: Addition `5 + 0.75` is done next, resulting in `5.75`.
3. `5 - 2 * 3 ** 4`
- Step 1: Exponentiation `3 ** 4` is done first, resulting in `81`.
- Step 2: Multiplication `2 * 81` is done next, yielding `162`.
- Step 3: Finally, the subtraction `5 - 162` results in `-157`.

## Question 5

1. Create a new variable `x`, and assign it the value 10.5.

2. Create a new variable `y`, and assign it the value 4.

3. Sum `x` and `y`, and make `x` refer to the resulting value. After this statement has been executed, what are the values of `x` and `y`?

- Step 1
``` Python
x = 10.5
```
- Step 2
``` Python
y = 4
```
- Step 3
``` Python
x = x + y
```
- Results
``` Python
x = 14.5
y = 4
```

## Question 6
Write a bullet list description of what happens when Python evaluates the statement `x += x - x` when x has the value 3.

- Step 1: Evaluate `x - x`, which is `3 - 3`, resulting in `0`.
- Step 2: Then, `x += 0` effectively adds `0` to the current value of `x`.
- Step 3: Finally, x still holds the value `3` because it was incremented by `0`.